delphi-checks/src/main/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheck.java

```diff
@@ -25,35 +25,40 @@
 import org.sonar.plugins.communitydelphi.api.reporting.QuickFix;
 import org.sonar.plugins.communitydelphi.api.reporting.QuickFixEdit;
 import org.sonar.plugins.communitydelphi.api.symbol.declaration.UnitImportNameDeclaration;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.UnitNameDeclaration;
 
 @Rule(key = "FullyQualifiedImport")
 public class FullyQualifiedImportCheck extends DelphiCheck {
 
   @Override
   public DelphiCheckContext visit(UnitImportNode unitImportNode, DelphiCheckContext context) {
-    if (!unitImportNode.isResolvedImport()) {
+    UnitImportNameDeclaration importDeclaration = unitImportNode.getImportNameDeclaration();
+    UnitNameDeclaration unitDeclaration = importDeclaration.getOriginalDeclaration();
+
+    if (unitDeclaration == null || isAliasImport(importDeclaration, unitDeclaration)) {
       return context;
     }
 
-    UnitImportNameDeclaration unitImportNameDeclaration = unitImportNode.getImportNameDeclaration();
-
-    String unitFullyQualifiedName = unitImportNameDeclaration.getOriginalDeclaration().getImage();
-    String unitImportName = unitImportNameDeclaration.getImage();
+    String actual = importDeclaration.fullyQualifiedName();
+    String expected = unitDeclaration.fullyQualifiedName();
 
-    if (unitImportName.length() != unitFullyQualifiedName.length()) {
+    if (!actual.equalsIgnoreCase(expected)) {
       context
           .newIssue()
           .onNode(unitImportNode)
           .withMessage(
-              "Fully qualify this unit name (found: \"%s\" expected: \"%s\").",
-              unitImportName, unitFullyQualifiedName)
+              "Fully qualify this unit name (found: \"%s\" expected: \"%s\").", actual, expected)
           .withQuickFixes(
               QuickFix.newFix("Fully qualify unit import")
-                  .withEdit(
-                      QuickFixEdit.replace(unitImportNode.getNameNode(), unitFullyQualifiedName)))
+                  .withEdit(QuickFixEdit.replace(unitImportNode.getNameNode(), expected)))
           .report();
     }
 
     return context;
   }
+
+  private static boolean isAliasImport(
+      UnitImportNameDeclaration importDeclaration, UnitNameDeclaration unitDeclaration) {
+    return !importDeclaration.simpleName().equalsIgnoreCase(unitDeclaration.simpleName());
```

Comment:
What if you alias `Scope.Name` to `Name`?

Reply:
Uh oh, this is a bit of an annoying headache to detect.
Guess it's time for a new API method, like `UnitImportNameDeclaration::isAliased`.